h/services/group_members.py

```diff
@@ -57,8 +61,21 @@ def get_memberships(
                 or_(GroupMembership.roles.contains(role) for role in roles)
             )
 
+        if offset is not None:
+            query = query.offset(offset)
+
+        if limit is not None:
+            query = query.limit(limit)
```

Comment:
There's an optimisation called "deferred joins" that a lot of pages about pagination mention, e.g. <https://planetscale.com/blog/mysql-pagination#deferred-joins-for-faster-offsetlimit-pagination>. But it's not clear how much benefit it would have in this particular case so I'm choosing to avoid premature optimisation. I plan to test the paginated API with large groups in production, if it turns out to be too slow we can consider any necessary optimisations.